developer/src/kmcmpdll/Compiler.cpp

```diff
@@ -561,7 +600,7 @@ BOOL CompileKeyboardHandle(HANDLE hInfile, PFILE_KEYBOARD fk)
   if (FSaveDebug) RecordDeadkeyNames(fk);
 
   /* Add the compiler version as a system store */
-  if ((msg = AddCompilerVersionStore(fk)) != CERR_None) SetError(msg);
+  if ((msg = AddStore(fk, TSS_COMPILEDVERSION, KEYMAN_VersionWin_W16)) != CERR_None) return msg;
 
```

Comment:
then we have to deal with     GetModuleFileName, GetFileVersionInfoSize, GetFileVersionInfo, VerQueryValue in GetFileVersion again:   https://github.com/keymanapp/keyman/pull/6980#issuecomment-1241103850

Reply:
AddCompilerVersionStore can use keymanversion.h to get the version information, so it can stay platform-agnostic. But we do need the additional functionality of the `-no-compiler-version` command line flag.

Reply:
👍